### ISSUE TITLE
AWS Solution Implementation: Search based on active field on patient api is not working

### ISSUE THREAD

**Issue by rye442** [awslabs/fhir-works-on-aws-deployment#227]
Search based on active field on patient api is not working and it is returning zero results even though active data are available.

## Create Patient Request
```
POST {{API_URL}}/Patient

{
  "resourceType": "Patient",
  "active": true,
  "name": [
    {
      "use": "official",
      "family": "442",
      "given": [
        "Rye"
      ]
    }
  ]
}
```

## Request
```
GET {{API_URL}}/Patient?active=true
```

## Results Body
```
{
  "resourceType": "Bundle",
  "id": "b6dd7abc-fe1f-405b-a603-a421470c810b",
  "meta": {
    "lastUpdated": "2021-03-01T05:22:58.206Z"
  },
  "type": "searchset",
  "total": 0,
  "link": [
    {
      "relation": "self",
      "url": "{{API_URL}}/Patient?active=true"
    }
  ],
  "entry": []
}
```

## Note

I am following [FHIR Works on AWS Implemation Guide](https://docs.aws.amazon.com/ja_jp/solutions/latest/fhir-works-on-aws/automated-deployment.html) and using CloudFormation to launch and use the template on the following URL.

https://s3.amazonaws.com/solutions-reference/fhir-works-on-aws/latest/fhir-works-on-aws.template

## Edit log
2021/3/3 - Added description of Request Body.

**Comment by rsmayda** [comment]
Thanks for reaching out! Could you provide an example Patient resource you POSTed to the server? It will help in reproducing the issue

**Comment by rsmayda** [comment]
I was able to search on `active` on the latest GitHub build will need to deploy the AWS Solution version to confirm if behavior is the same there

**Comment by Bingjiling** [comment]
Hi @rye442 , thanks for reaching out! I deployed AWS Solutions and was able to recreate the bug. AWS Solution build is not up to date with the GitHub build, in particular, this [update](https://github.com/awslabs/fhir-works-on-aws-search-es/pull/36) was not included, causing error with search parameter `active`. It is on our radar to bring the latest codes into AWS Solution build, however, we don't have an estimate on it yet. I would recommend using the latest GitHub install to avoid running into this issue. Hope it helps.

**Comment by rye442** [comment]
Hi @rsmayda, Thank you for your quick reply and sharing the reproduced results in the GitHub build version!
As you requested, I have added Patient's example to this issue.

**Comment by rye442** [comment]
Hi @Bingjiling, thank you for your prompt reply and sharing the results of reproducing the code with the AWS solution build.

I understand that I should use the latest GitHub Build (which reflects [this bug fix](https://github.com/awslabs/fhir-works-on-aws-search-es/pull/36)). I am currently using Windows, am I correct in understanding that in this case I can build on AWS using the method described on [this page](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/INSTALL.md#windows-installation)?
I will try as soon as I can (within the next week), but it may be later due to other work commitments.

I feel that it is highly undesirable for such a bug to exist in the AWS solution version. I hope that this update will be reflected in the AWS Solution Edition as soon as possible.

**Comment by Bingjiling** [comment]
Hi @rye442, thanks for the update! 

Yes, you can follow [this page](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/INSTALL.md#windows-installation) to deploy in Windows. And please let us know if you run into any issues. 

I'll bring the request for updating AWS solution version to the team and see the nearest timeline we can fit it in. 

Thanks again for your interest in FHIRWorks.

**Comment by rye442** [comment]
Hi @Bingjiling, thanks for the reply.

I tried the Windows Installation but was thwarted by two problems. I have already registered an issue for this as #233 and #234.

I will continue to check the status of AWS solution version in this issue.

**Comment by SanketD92** [comment]
Hello @rye442, since the issues you mentioned in your last response are now closed I wanted to touch base and see if you're unblocked to use FHIRWorks per your use-case.

Meanwhile, the bug for fixing this in the Solutions version of the product will be prioritized by the team soon, as @Bingjiling mentioned.

**Comment by rye442** [comment]
Hello @SanketD92, currently, I'm able to build FHIR Works with Windows Installation.

On the other hand, the version provided as an implementation of the AWS solution does not have the basic functionality shown as this issue, which caused my team to have strong doubts.

Therefore, I feel that it is very good news to give priority to this issue.

**Comment by masterelaya** [comment]
I am getting boto3 error when install the fhir on AWS through docker. Please help.
#6 31.75 Collecting boto3
#6 31.93   Could not find a version that satisfies the requirement boto3 (from versions: )
#6 31.93 No matching distribution found for boto3

RUN apt-get update && apt-get install -yq python3-pip less && pip3 install boto3 && curl -fsSL --compressed  "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o /tmp/awscliv2.zip && unzip -q /tmp/awscliv2.zip -d /tmp && /tmp/aws/install && npm install serverless -g:

**Comment by Bingjiling** [comment]
Fhir Works on AWS Solutions has released [v4.0.0-Solution](https://github.com/awslabs/fhir-works-on-aws-deployment/releases/tag/v4.0.0-Solution), which is a functional equivalent to github v4.0.0. Please follow the implementation guide on the [Solutions landing page](https://aws.amazon.com/solutions/implementations/fhir-works-on-aws/) for more information.
